app/models/katello/concerns/organization_extensions.rb

```diff
@@ -228,6 +228,11 @@ def expiring_subscriptions
           subscriptions.select(&:expiring_soon?)
         end
 
+        def clear_syspurpose_status
+          host_purpose = HostStatus::Status.where(type: 'Katello::PurposeStatus')
```

Comment:
This is going to clear out the system purpose status for all hosts - in every organization!

You can limit to the relevant org like this:

```
HostStatus::Status.where(type: 'Katello::PurposeStatus', host: organization.hosts)
```

The issue we've seen with queries like this is that postgresql will not like when there are a LOT of host ids in the subquery, so we can form it a bit differently to avoid that:

```
HostStatus::Status.where(type: 'Katello::PurposeStatus').where('host_id in (?)', organization.hosts.pluck(:id))
```

That said, I think we should clear out all of the other purpose status types since they'd just be hanging around for no reason: PurposeRoleStatus, PurposeSlaStatus, PurposeAddonsStatus. You can pass multiple types as an array in your query!

Of course, you should verify that everything I said is accurate :+1:

Reply:
Good catch on that, I was able to find the correct way to delete all of the stuff I need to and use your way with a change in the hosts select statement using self.hosts to get the hosts.